source/plugins/commands.rst

```diff
@@ -2,3 +2,76 @@
 Command Registration
 ====================
 
+Create a command
+================
+
+The first step is to create a class for the command. The class has to implement ``CommandCallable``:
+
+.. code-block:: java
+
+    package com.github.maenni025.spongeExample;
+
+    import java.util.Collections;
+    import java.util.List;
+
+    import org.spongepowered.api.Server;
+    import org.spongepowered.api.entity.player.Player;
+    import org.spongepowered.api.util.command.CommandCallable;
+    import org.spongepowered.api.util.command.CommandException;
+    import org.spongepowered.api.util.command.CommandSource;
+
+    import com.google.common.base.Optional;
+
+    public class ExampleCommand implements CommandCallable {
+
+        private final Server server;
+
+        public ExampleCommand(Server server) {
+            this.server = server;
+        }
+
+        public boolean call(CommandSource source, String arguments, List<String> parents) throws CommandException {
+            for (Player p : server.getOnlinePlayers()) {
+                p.sendMessage(arguments);
+            }
+            return true;
+        }
+
+        public boolean testPermission(CommandSource source) {
+            return source.hasPermission("example.exampleCommand");
+        }
+
+        public Optional<String> getShortDescription() {
+            Optional<String> desc = Optional.of("Displays a message to all players");
```

Comment:
You could return these directly, or store them in a field, so you don't have to initialize a new Optional every time.